### PR TITLE
#24 - adding the backend code to support deploying a user's modules

### PR DESCRIPTION
**Pull request by paxtonhare**
- also added domains to the Java API and xquery

### CODE REVIEW

File: data-hub/src/hub-in-a-box/root/com.marklogic.hub/lib/flow-lib.xqy [hunk 1]
```diff
@@ -274,7 +409,14 @@ declare function flow:run-flow(
     ()
 };
 
+(:~
+ : Construct an envelope
+ :
+ : @param $map - a map with all the stuff in it
+ : @return - the newly constructed envelope
+ :)
 declare function flow:make-envelope($map as map:map)
```
Comment (dmcassel) [inline data-hub/src/hub-in-a-box/root/com.marklogic.hub/lib/flow-lib.xqy]: Envelope is one of your transforms, right? I get that you're providing a built-in one, but would it be helpful to put this in a separate file to illustrate how to build other ones?

File: data-hub/src/hub-in-a-box/services/metadata/domain.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!-- Copyright 2016 MarkLogic Corporation.  All Rights Reserved. -->
```
Comment (dmcassel) [inline data-hub/src/hub-in-a-box/services/metadata/domain.xml]: Would be cool if (at some point) you had a scriptable test to compare this to what the code provides, to make sure they stay in sync. Might not be easy, though.

File: data-hub/src/main/java/com/marklogic/hub/flow/SimpleFlow.java [hunk 3]
```diff
@@ -26,8 +26,8 @@
  */
 public class SimpleFlow extends AbstractFlow {
 
-    public SimpleFlow(String name) {
-        super(name, "simple");
+    public SimpleFlow(String domainName, String name) {
```
Comment (dmcassel) [inline data-hub/src/main/java/com/marklogic/hub/flow/SimpleFlow.java]: would be better to use "flowName", I think -- also consistent with FlowManagerService.

File: data-hub/src/hub-in-a-box/root/com.marklogic.hub/lib/flow-lib.xqy (hunk 1 above)
Comment (paxtonhare) [inline data-hub/src/hub-in-a-box/root/com.marklogic.hub/lib/flow-lib.xqy]: Actually it's not a built in transform. that concept is so last week :)

we are not allowing the user to override the envelope.

**Pull request merged**